### PR TITLE
entt: init at 3.5.2

### PR DESCRIPTION
**Pull request by Twey**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/entt/default.nix [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+stdenv.mkDerivation rec {
+  name = "entt";
+  version = "3.5.2";
+  src = fetchFromGitHub {
+    owner = "skypjack";
+    repo = "entt";
+    rev = "v${version}";
+    sha256 = "1p09p1wn8cbj17z83iyyy2498wy1gzyi2mmqi5i2cxglslbm6hy0";
+  };
+  buildInputs = [ cmake ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/entt/default.nix:11]:
```suggestion
  version = "3.5.2";

  src = fetchFromGitHub {
    owner = "skypjack";
    repo = "entt";
    rev = "v${version}";
    sha256 = "1p09p1wn8cbj17z83iyyy2498wy1gzyi2mmqi5i2cxglslbm6hy0";
  };

  nativeBuildInputs = [ cmake ];
```

File: pkgs/development/libraries/entt/default.nix [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{ stdenv, fetchFromGitHub, cmake }:
+stdenv.mkDerivation rec {
+  name = "entt";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/entt/default.nix:3]:
```suggestion
  pname = "entt";
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102036` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>entt</li>
  </ul>
</details>

**Comment by Twey** [comment]
I don't understand this error.  This looks like CMake is generating Makefiles that depend on `/bin/sh`?  `/bin/sh` isn't mentioned in the CMake of the source.

**Comment by Twey** [comment]
(thank you for the suggestions!)

File: pkgs/development/libraries/entt/default.nix [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    owner = "skypjack";
+    repo = "entt";
+    rev = "v${version}";
+    sha256 = "1p09p1wn8cbj17z83iyyy2498wy1gzyi2mmqi5i2cxglslbm6hy0";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  meta = with stdenv.lib; {
+    homepage = https://github.com/skypjack/entt;
```
Comment (Flakebi) [inline pkgs/development/libraries/entt/default.nix:16]: Urls should be quoted (I think the plan is to deprecate the syntax for unquoted urls)

**Comment by Flakebi** [comment]
Result of `nixpkgs-review pr 102036` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>entt</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102036` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>entt</li>
  </ul>
</details>

**Review by Flakebi — APPROVED** [review]

File: pkgs/development/libraries/entt/default.nix [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    sha256 = "1p09p1wn8cbj17z83iyyy2498wy1gzyi2mmqi5i2cxglslbm6hy0";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  meta = with stdenv.lib; {
+    homepage = "https://github.com/skypjack/entt";
+    description = "A header-only, tiny and easy to use library for game programming and much more written in modern C++";
+    maintainers = with maintainers; [ twey ];
+    license = licenses.mit;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/entt/default.nix:19]: Can you please add platforms.

Comment (doronbehar) [reply pkgs/development/libraries/entt/default.nix:19]: No platforms specified means I guess all platforms should be supported.

**Review by doronbehar — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102036` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>entt</li>
  </ul>
</details>

**Pull request merged**